### DISCUSSION TITLE
HomeProxy的DNS配置CN规则集后，会DNS泄漏，去掉就无问题

### DISCUSSION THREAD

**Discussion by careysucci** [immortalwrt/homeproxy#217] [Q&A]
CN规则集连接：https://github.com/SagerNet/sing-geoip/raw/refs/heads/rule-set/geoip-cn.srs

自建两个DNS Server（AdguardHome）：
国外DNS: A
国内DNS: B
DNS默认出口是: A
路由直连规则：按规则集CN匹配默认出口直连

dns规则里配置CN DNS解析规则：规则集选择CN规则--dns解析选择国内DNS服务器B
问题场景描述：
1. 当存在CN这条DNS规则时，会把国外site的dns解析导到DNS服务器B
2. 当删除CN这条DNS规则时，就不会存在DNS泄漏了
3. 路由规则也配置CN规则，但不会出现国外数据直连的情况，只有DNS解析会有这种情况

问题：
1. 如果是个人配置问题，请大佬帮指正哪些地方需要注意及检查修改
2. 如果是homeproxy bug请问会有大佬关注么？

**Comment by 1715173329** [comment]
购买检测“泄漏”网站的 VPN 即可，立马 250% 安全。
